analyses/tp53_nf1_score/04-tp53-altered-annotation.Rmd

```diff
@@ -0,0 +1,355 @@
+  mutate(
+    # add tp53_altered annotation column
+    tp53_altered = 
+      case_when(
+        # when activating == ""
+        activating == "" &
+          # check if mutated variant AA position overlaps hotspot database 
+          ( hotspot == "Yes"  |
+              # check if sample_id has SNV+CNV mutation
+              # suggesting both alleles are mutated
+              (SNV_indel_counts >= 1 & CNV_loss_counts >=1) |
+              # check if more than 1 SNV is present
+              # suggesting both alleles are mutated
+              SNV_indel_counts >= 1 |
+              # check if SNV mutant and has
+              # Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (SNV_indel_counts >= 1 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions)) |
+              # check if CNV mutant  and has
+              # Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (CNV_loss_counts >= 1 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions)) |
+              # check if tp53 inactivating score for RNA_Seq is greater that 0.5
+              # and has Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (tp53_score > 0.5 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions)) |
+              # check if tp53 inactivating score for RNA_Seq is greater that 0.5
+              # and has 1 or more SNV| CNV loss in TP53
+              (tp53_score > 0.5 & 
+                             (SNV_indel_counts >= 1 | CNV_loss_counts >=1))
+          ) ~ "loss",
+        # when activating == "Yes"
+        activating == "Yes" ~ "activated",
+        # if no evidence supports TP53 inativation or activation 
+        TRUE ~ "Other"
+      )
+  )
+
+```
+
+## Explore distribution of tp53_altered status vs tp53 inactivation scores
+
+```{r}
+ggplot(tp53_alterations_score_wide, aes(x = factor(tp53_altered), y = tp53_score)) +
+  geom_violin()+
+  geom_jitter(alpha = 0.5, width = 0.2) +
+  stat_compare_means() +
+  theme_bw() +
+  ggtitle("Distribution of scores across tp53 altered status") +
+  theme(axis.text.x = element_text(angle = 60, hjust = 1))+
+  xlab("tp53 altered status") 
+
+```
+TP53 mutants with gain-of-function (activating) mutants promote tumorigenesis according to literature. [Reference](https://pubmed.ncbi.nlm.nih.gov/17417627/) and [reference](https://pubmed.ncbi.nlm.nih.gov/24677579/) have similar distribution of classifier scores to that of potential loss-of-function (multi-allelic) Tp53 mutations. 
+
+"Other" annotated samples either don't have any high-confidence loss/gain TP53 SNVs nor CNV losses OR DNA sample is not available.
+
+### (stranded) Expression profile for activating vs loss TP53 status
+Using stranded because there are more samples in stranded
```

Comment:
Will you also plot polyA here (in a separate plot - you can facet library), even though there are fewer samples?

Reply:
done!